packages/ketcher-react/src/script/ui/views/toolbars/ToolbarGroupItem/ToolbarMultiToolItem/ToolbarMultiToolItem.tsx

```diff
@@ -0,0 +1,144 @@
+import { usePortalOpening } from './usePortalOpening'
+import { usePortalStyle } from './usePortalStyle'
+import { chooseMultiTool } from './variants/chooseMultiTool'
+import { MultiToolVariant } from './variants/variants.types'
+
+import classes from './ToolbarMultiToolItem.module.less'
+
+interface ToolbarMultiToolItemProps {
+  id: ToolbarItemVariant
+  options: ToolbarItem[]
+  groups?: number[]
+  variant?: MultiToolVariant
+  status: {
+    [key in string]?: UiAction
+  }
+  opened: string | null
+  disableableButtons: string[]
+  indigoVerification: boolean
+  className?: string
+}
+
+interface ToolbarMultiToolItemCallProps {
+  onAction: (action: UiActionAction) => void
+  onOpen: (menuName: string, isSelected: boolean) => void
+}
+
+type Props = ToolbarMultiToolItemProps & ToolbarMultiToolItemCallProps
+
+const ToolbarMultiToolItem = (props: Props) => {
+  const {
+    id,
+    options,
+    groups,
+    variant,
+    status,
+    opened,
+    indigoVerification,
+    disableableButtons,
+    className,
+    onAction,
+    onOpen
+  } = props
+
+  const ref = useRef<HTMLDivElement>(null)
+  const [isOpen] = usePortalOpening([id, opened, options])
+  const [portalStyle] = usePortalStyle([ref, isOpen])
+
+  let selected = false
+  let currentId = id
+
+  const selectedTool = options.find(
+    toolbarItem => status[toolbarItem.id]?.selected
+  )
+  if (selectedTool) {
+    currentId = selectedTool.id
+    selected = true
+  }
+
+  const currentStatus = status[currentId]
+  // todo: #type find out real type, possible GetActionState is no acceptable here
+  // and check this type convert is redundant
+  selected = selected || Boolean(currentStatus?.selected)
+
+  if (!currentStatus && options.length) {
+    currentId = options[0].id
+  }
+
+  const actionButtonProps: Omit<
+    ActionButtonProps & ActionButtonCallProps,
+    'name' | 'status' | 'action'
+  > = {
+    disableableButtons,
+    indigoVerification,
+    onAction
+  }
+
+  const onOpenOptions = () => {
+    // todo: same as #type above
+    onOpen(id, Boolean(currentStatus?.selected))
+  }
+
+  const [Component, portalClassName] = chooseMultiTool(variant)
+
+  return (
+    <div ref={ref} className={classes.root}>
+      <ActionButton
+        {...actionButtonProps}
+        className={className}
+        name={currentId}
+        action={action[currentId]}
+        // @ts-ignore
+        status={currentStatus}
+        selected={selected}
+      />
+      <Icon className={classes.icon} name="dropdown" onClick={onOpenOptions} />
+
+      {!isOpen ? null : (
+        <Portal
+          isOpen={isOpen}
+          className={clsx(classes.portal, portalClassName)}
+          style={portalStyle}>
+          <Component
+            options={options}
+            groups={groups}
+            status={status}
+            disableableButtons={disableableButtons}
+            indigoVerification={indigoVerification}
+            onAction={onAction}
+          />
+        </Portal>
+      )}
```

Comment:
It's a bit easier to read code without NOT operator:
```suggestion
      {isOpen ? (
        <Portal
          isOpen={isOpen}
          className={clsx(classes.portal, portalClassName)}
          style={portalStyle}>
          <Component
            options={options}
            groups={groups}
            status={status}
            disableableButtons={disableableButtons}
            indigoVerification={indigoVerification}
            onAction={onAction}
          />
        </Portal>
      ) : null}
```

Reply:
fixed in [code review fixes 2](https://github.com/epam/ketcher/pull/318/commits/518fd1d7ccaedaf6220eac0409121397e4f8fe0d)